src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs

```diff
@@ -402,34 +403,31 @@ public async Task<IActionResult> ExternalLoginCallback(string returnUrl = null,
             }
             else
             {
-                var email = info.Principal.FindFirstValue(ClaimTypes.Email) ?? info.Principal.FindFirstValue("email");
-
-                if (!string.IsNullOrWhiteSpace(email))
+                if (_identityOptions.User.RequireUniqueEmail)
                 {
-                    iUser = await _userManager.FindByEmailAsync(email);
-                }
+                    iUser = await _userManager.FindByEmailAsync(info.GetEmail());
```

Comment:
Only this should be in the `if`. Otherwise, it would require a unique e-mail even for accounts found by `FindByLoginAsync()` above.

Reply:
I have no problem to make the change but if `FindByLoginAsync()` find a User, this code can't be execute.

https://github.com/OrchardCMS/OrchardCore/blob/3853b973f59196c0f5c75a6c72b885679b5bbc8c/src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs#L382-L403

fixed in f2faf17

Reply:
I'm not sure I understand. If `FindByLoginAsync()` finds a user, then the code block you show will execute, and rightfully so. In your case, that would mean that their PLUTO ID matched, what means that they're definitely the same user. In your use case, it won't find a user when you want the second account with the same e-mail to log in. And you'll need to configure `RequireUniqueEmail` as `false` too.